### PR TITLE
add typ values

### LINKED ISSUE #30: typ values for Client Attestation JWT and Client Attestation PoP JWT
The current draft does not require the Client Attestation JWT and Client Attestation PoP JWT to have the `typ` JWS header parameter. Do you have any plans to define specific `typ` values for these JWTs?

### LINKED ISSUE #72: Add required typ values for the client-attestation and client-attestation-pop in line with JWT BCP

### PR DESCRIPTION
**Pull request by paulbastian**
Closes #30 
Closes #72

### CODE REVIEW

**Comment by c2bo** [comment]
This includes #90 which makes it pretty hard to see the changes to main :/ I guess let's wait until 90 is merged?

File: draft-ietf-oauth-attestation-based-client-auth.md
Comment (c2bo) [inline draft-ietf-oauth-attestation-based-client-auth.md:146]: Should we explain what happens if claims are not understood by a RP?
We have 2 options and should probably clearly state how to deal with such a situation in the spec:
- ignore not understood claims
- fail validation of the client attestation

File: draft-ietf-oauth-attestation-based-client-auth.md
Comment (c2bo) [inline draft-ietf-oauth-attestation-based-client-auth.md:198]: Same as above: How to deal with claims that are not understood by a RP?

File: draft-ietf-oauth-attestation-based-client-auth.md
Comment (c2bo) [reply draft-ietf-oauth-attestation-based-client-auth.md:157]:
```suggestion
1. The JWT MAY contain other claims. All claims that are not understood by implementations MUST be ignored.
```

File: draft-ietf-oauth-attestation-based-client-auth.md
Comment (c2bo) [reply draft-ietf-oauth-attestation-based-client-auth.md:209]:
```suggestion
1. The JWT MAY contain other claims. All claims that are not understood by implementations MUST be ignored.
```

**Review by c2bo — APPROVED** [review]

**Review by tplooker — APPROVED** [review]

**Pull request merged**
